tests/test_dataset.py

```diff
@@ -1396,6 +1409,68 @@ def test_adding_layer_with_valid_dtype_per_layer() -> None:
     ds.add_layer("color2", Layer.COLOR_TYPE, dtype_per_layer=np.uint8, num_channels=1)
     ds.add_layer("color3", Layer.COLOR_TYPE, dtype_per_channel=np.uint8, num_channels=3)
     ds.add_layer("color4", Layer.COLOR_TYPE, dtype_per_channel="uint8", num_channels=3)
+    ds.add_layer(
+        "seg1",
+        Layer.SEGMENTATION_TYPE,
+        dtype_per_channel="float",
+        num_channels=1,
+        largest_segment_id=100000,
+    )
+    ds.add_layer(
+        "seg2",
+        Layer.SEGMENTATION_TYPE,
+        dtype_per_channel=np.float,
+        num_channels=1,
+        largest_segment_id=100000,
+    )
+    ds.add_layer(
+        "seg3",
+        Layer.SEGMENTATION_TYPE,
+        dtype_per_channel=float,
+        num_channels=1,
+        largest_segment_id=100000,
+    )
+    ds.add_layer(
+        "seg4",
+        Layer.SEGMENTATION_TYPE,
+        dtype_per_channel="double",
+        num_channels=1,
+        largest_segment_id=100000,
+    )
+    ds.add_layer(
+        "seg5",
+        Layer.SEGMENTATION_TYPE,
+        dtype_per_channel="float",
+        num_channels=3,
+        largest_segment_id=100000,
+    )
+
+    with open(TESTOUTPUT_DIR / "valid_dtype" / "datasource-properties.json", "r") as f:
+        data = json.load(f)
+        # The order of the layers in the properties equals the order of creation
+        assert data["dataLayers"][0]["elementClass"] == "uint24"
```

Comment:
@fm3 Should we deprecate `uint24` in favor of `uint8x3`? Same for `float96` -> `float32x3` (not sure if wk supports this).

Reply:
wK does not currently support the `x3` syntax. I agree that it would be easier to read, but I am not sure it’s worth the effort, since legacy properties jsons would still have to be supported also, creating more complexity.